pyfr/integrators/writers.py

```diff
@@ -22,13 +22,67 @@ def __init__(self, *args, **kwargs):
         # Output counter (incremented each time output() is called)
         self.nout = 0
 
+        # MPI info
+        comm, rank, root = get_comm_rank_root()
+
+        # Get the type and shape of each element in the partition
+        etypes, shapes = self._system.ele_types, self._system.ele_shapes
+
+        # Gather this information onto the root rank
+        eleinfo = comm.gather(zip(etypes, shapes), root=root)
+
+        # Deciding if parallel
+        parallel = h5py.get_config().mpi
+        parallel &= h5py.version.version_tuple[:2] >= (2, 5)
+        parallel &= not self.cfg.getbool('soln-output', 'serial-h5', False)
+
+        if parallel:
```

Comment:
Merge if/else block with that on line 44 (the condition `if parallel` is the same for both blocks).